client/src/SubmitBidForm.js

```diff
@@ -43,29 +40,45 @@ const SubmitBidForm = () => {
 	const handleSubmit = async (e) => {
 		e.preventDefault();
 
+		const newErrors = [];
+
+		if (coverLetter.length > 1000) {
+			newErrors.push("Maximum length is upto 1,000 characters");
+		}
+
 		const duration = parseInt(proposedDuration);
+		if (isNaN(duration) || duration < 1 || duration > 1000) {
+			newErrors.push("Duration must be between 1 and 1,000 days");
+		}
+
 		const budget = parseFloat(proposedBudget);
-		try {
-			const bidData = {
-				tenderId,
-				bidding_amount: budget,
-				cover_letter: coverLetter,
-				suggested_duration_days: duration,
-				bidding_date: new Date(),
-			};
-
-			await post("/api/bid", bidData);
-
-			alert("Bid submitted successfully!");
-			navigate("/dashboard");
-		} catch (error) {
-			const { status, data } = error.response;
-			if (status === 400) {
-				setRegisterStatus("Validation error");
-				setValidationErrors(data.errors);
-			} else {
-				setRegisterStatus("A server error occurred. Please try again later.");
-				setValidationErrors(data.errors);
+		if (isNaN(budget) || budget <= 0) {
+			newErrors.push("Input a valid bidding amount");
+		}
+
+		if (newErrors.length > 0) {
+			setErrors(newErrors);
+		} else {
+			try {
+				const bidData = {
+					tenderId,
+					bidding_amount: budget,
+					cover_letter: coverLetter,
+					suggested_duration_days: duration,
+					bidding_date: new Date(),
+				};
+
+				await post("/api/bid", bidData);
+
+				alert("Bid submitted successfully!");
+				navigate("/dashboard");
+			} catch (error) {
+				const { status, data } = error.response;
+				if (status === 400) {
+					setErrors(data.errors);
+				} else {
+					setErrors(data.errors);
```

Comment:
Please, test your code. This doesn't work. `errors` is only defined for `400` errors